core/lib/db_connection/src/instrument.rs

```diff
@@ -527,3 +527,30 @@ mod tests {
             .unwrap();
     }
 }
+
+pub trait DalContext<T> {
+    /// Wraps a provided argument validation error. It is assumed that the returned error
+    /// will be returned as an error cause (e.g., it is logged as an error and observed using metrics).
+    fn arg<Q>(self, i: &Instrumented<Q>, arg_name: &'static str) -> DalResult<T>;
+}
+
+impl<T, E> DalContext<T> for Result<T, E>
+where
+    E: Into<anyhow::Error>,
+{
+    fn arg<Q>(self, i: &Instrumented<Q>, arg_name: &'static str) -> DalResult<T> {
```

Comment:
- Isn't using `Instrumented<'_, ()>` sufficient here? AFAIU, the query is always bound later.
- Bikeshedding: `arg` is an overly generic name; looking at it in isolation, I would think it would be an argument getter or something similar. If we go by analogy with `SqlxContext`, it should be called something like `encode_arg`.